src/go/rpk/pkg/config/config_test.go

```diff
@@ -59,125 +57,107 @@ func TestSet(t *testing.T) {
-				require.Exactly(st, 42.3, mgr.v.Get("redpanda.float_field"))
+			check: func(st *testing.T, cfg *Config) {
+				require.Exactly(st, 42.3, cfg.Redpanda.Other["float_field"])
 			},
 		},
 		{
-			name:   "it should set single string fields",
+			name:   "set single string fields",
 			key:    "redpanda.data_directory",
-			value:  "'/var/lib/differentdir'",
+			value:  "/var/lib/differentdir",
 			format: "single",
-			check: func(st *testing.T, c *Config, _ *manager) {
-				require.Exactly(st, "'/var/lib/differentdir'", c.Redpanda.Directory)
+			check: func(st *testing.T, c *Config) {
+				require.Exactly(st, "/var/lib/differentdir", c.Redpanda.Directory)
 			},
 		},
 		{
-			name:  "it should detect single string fields if format isn't passed",
+			name:  "detect single string fields if format isn't passed",
 			key:   "redpanda.data_directory",
 			value: "/var/lib/differentdir",
-			check: func(st *testing.T, c *Config, _ *manager) {
+			check: func(st *testing.T, c *Config) {
 				require.Exactly(st, "/var/lib/differentdir", c.Redpanda.Directory)
 			},
 		},
 		{
-			name:   "it should set single bool fields",
+			name:   "set single bool fields",
 			key:    "rpk.enable_usage_stats",
 			value:  "true",
 			format: "single",
-			check: func(st *testing.T, c *Config, _ *manager) {
+			check: func(st *testing.T, c *Config) {
 				require.Exactly(st, true, c.Rpk.EnableUsageStats)
 			},
 		},
 		{
-			name:  "it should detect single bool fields if format isn't passed",
+			name:  "detect single bool fields if format isn't passed",
 			key:   "rpk.enable_usage_stats",
 			value: "true",
-			check: func(st *testing.T, c *Config, _ *manager) {
+			check: func(st *testing.T, c *Config) {
 				require.Exactly(st, true, c.Rpk.EnableUsageStats)
 			},
 		},
 		{
-			name:   "it should partially set map fields (yaml)",
-			key:    "rpk",
-			value:  `tune_disk_irq: true`,
+			name: "partially set map fields (yaml)",
+			key:  "rpk",
+			value: `tune_disk_irq: true
+tune_cpu: true`,
 			format: "yaml",
-			check: func(st *testing.T, c *Config, _ *manager) {
-				expected := RpkConfig{
-					EnableUsageStats:         false,
-					Overprovisioned:          false,
-					TuneNetwork:              false,
-					TuneDiskScheduler:        false,
-					TuneNomerges:             false,
-					TuneDiskIrq:              true,
-					TuneCPU:                  false,
-					TuneAioEvents:            false,
-					TuneClocksource:          false,
-					TuneSwappiness:           false,
-					TuneTransparentHugePages: false,
-					EnableMemoryLocking:      false,
-					TuneFstrim:               false,
-					TuneCoredump:             false,
-					TuneDiskWriteCache:       false,
-					CoredumpDir:              "/var/lib/redpanda/coredump",
-				}
-				require.Exactly(st, expected, c.Rpk)
+			check: func(st *testing.T, c *Config) {
+				require.Exactly(st, true, c.Rpk.TuneDiskIrq)
+				require.Exactly(st, true, c.Rpk.TuneCPU)
```

Comment:
Can this check one more config knob is false? The benefit of the old test is that it ensures only two fields are set to true -- this new test loses the full check. Is this full check anywhere else? If so, does this test here provide any benefit?